### DISCUSSION TITLE
API: watch() function should return not only value but also other field statuses

### DISCUSSION THREAD

**Discussion by anhquande** [react-hook-form/react-hook-form#2570] [Ideas] [answered]
The watch() function currently (version 6.3.1) returns only the value.

It maybe better if it can return other field statuses, e.g. dirty, error.

What do you think about it?

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the suggestion, formState is already optimized, eg: if you are not reading `isValid`, all the validation logic will be skipped and extra re-render will be avoided too.

**Reply by anhquande** [reply]
thanks @bluebill1049 for your quick response. I find your library very cool, and your support is unbelievably fast

**Reply by bluebill1049** [reply]
thank you. I try my best to support developers around the world :)
